torch/csrc/utils/python_arg_parser.cpp

```diff
@@ -111,6 +113,9 @@ bool FunctionParameter::check(PyObject* obj) {
     case ParameterType::PYOBJECT: return true;
     case ParameterType::DTYPE: return THPDtype_Check(obj);
     case ParameterType::LAYOUT: return THPLayout_Check(obj);
+    case ParameterType::DEVICE:
+      return THPUtils_checkLong(obj) || THPUtils_checkString(obj) || THPDevice_Check(obj);
```

Comment:
nit: every check costs something, and they have to be performed in this order. I'd vote to use an order of:
1. Device
2. String
3. Long (legacy)

This will ensure that the recommended Device path can dispatch quickly.

Reply:
I put long first for this reason because all the current usages use longs, so it will be the most common.  Also thought string would be more common than device, because most people probably just do `device='cuda'` or whatever.

Reply:
Alright, let's leave it